### PR TITLE
HomeScreen Enhancement

### PR DESCRIPTION
**Pull request by mujafar7**
Screenshot_20231002_211817
Screenshot_20231002_211812
Screenshot_20231002_211807
Screenshot_20231002_211802
Screenshot_20231002_211915
Screenshot_20231002_211905
Screenshot_20231002_211832
Screenshot_20231002_211821

### CODE REVIEW

File: .env.local.example
Comment (Sutara-Skanda) [inline .env.local.example:1]: don't push env file revert this commit and create PR again

**Review by bhanugummana — CHANGES_REQUESTED** [review]

**Review by Sutara-Skanda — CHANGES_REQUESTED** [review]

File: .env.local.example [hunk 1]
```diff
@@ -3,8 +3,8 @@ REACT_APP_AUTH_DOMAIN=
 REACT_APP_PROJECT_ID=
 REACT_APP_STORAGE_BUCKET=
 REACT_APP_MESSAGING_SENDER_ID=
-REACT_APP_APP_ID=
-REACT_APP_MEASUREMENT_ID=
+REACT_APP_APP_ID=1:
```
Comment (bhanugummana) [inline .env.local.example:6]: remove 1 here

File: src/pages/DisplayChapters.tsx [hunk 2]
```diff
@@ -0,0 +1,439 @@
...
+      onCourseChanges(getCourseByUrl);
+    }
+  }, [getCourseByUrl]);
+  useEffect(() => {
+    console.log("chapters", currentCourse);
+    console.log("local grade map", localGradeMap);
+    if (!localGradeMap || !localGradeMap.grades) {
+      if (currentCourse) {
+        setIsLoading(true);
+        api.getDifferentGradesForCourse(currentCourse).then(({ grades }) => {
```
Comment (bhanugummana) [inline src/pages/DisplayChapters.tsx:83]: use await instead of then

File: src/pages/DisplayChapters.tsx [hunk 3]
```diff
@@ -0,0 +1,439 @@
...
+    const currentStudent = await Util.getCurrentStudent();
+    if (!currentStudent) {
+      // history.replace(PAGES.DISPLAY_STUDENT);
+      history.replace(PAGES.SELECT_MODE);
+      return [];
+    }
+    // const currClass = localStorage.getItem(CURRENT_CLASS);
+    const currClass = schoolUtil.getCurrentClass();
+    if (!!currClass) setCurrentClass(currClass);
+    api.getStudentResultInMap(currentStudent.docId).then(async (res) => {
```
Comment (bhanugummana) [inline src/pages/DisplayChapters.tsx:238]: use await instead of then

File: src/pages/Subjects.tsx [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+    const currentStudent = await Util.getCurrentStudent();
+    if (!currentStudent) {
+      // history.replace(PAGES.DISPLAY_STUDENT);
+      history.replace(PAGES.SELECT_MODE);
+      return [];
+    }
+    // const currClass = localStorage.getItem(CURRENT_CLASS);
+    const currClass = schoolUtil.getCurrentClass();
+    if (!!currClass) setCurrentClass(currClass);
+    api.getStudentResultInMap(currentStudent.docId).then(async (res) => {
```
Comment (bhanugummana) [inline src/pages/Subjects.tsx:206]: use await instead of then

**Review by bhanugummana — CHANGES_REQUESTED** [review]

**Comment by mujafar7** [comment]
Closing the PR As need to work on request changes

**Pull request closed without merging**